include/crocoddyl/multibody/actions/contact-fwddyn.hpp

```diff
@@ -19,49 +19,76 @@
 
 namespace crocoddyl {
 
-class DifferentialActionModelContactFwdDynamics : public DifferentialActionModelAbstract {
+template <typename _Scalar>
+class DifferentialActionModelContactFwdDynamicsTpl : public DifferentialActionModelAbstractTpl<_Scalar> {
  public:
-  DifferentialActionModelContactFwdDynamics(boost::shared_ptr<StateMultibody> state,
-                                            boost::shared_ptr<ActuationModelFloatingBase> actuation,
-                                            boost::shared_ptr<ContactModelMultiple> contacts,
-                                            boost::shared_ptr<CostModelSum> costs, const double& JMinvJt_damping = 0.,
-                                            const bool& enable_force = false);
-  ~DifferentialActionModelContactFwdDynamics();
-
-  void calc(const boost::shared_ptr<DifferentialActionDataAbstract>& data, const Eigen::Ref<const Eigen::VectorXd>& x,
-            const Eigen::Ref<const Eigen::VectorXd>& u);
-  void calcDiff(const boost::shared_ptr<DifferentialActionDataAbstract>& data,
-                const Eigen::Ref<const Eigen::VectorXd>& x, const Eigen::Ref<const Eigen::VectorXd>& u);
+  typedef _Scalar Scalar;
+  typedef DifferentialActionModelAbstractTpl<Scalar> Base;
+  typedef MathBaseTpl<Scalar> MathBase;
+  typedef CostModelSumTpl<Scalar> CostModelSum;
+  typedef StateMultibodyTpl<Scalar> StateMultibody;
```

Comment:
Add pinocchio typedef
```cpp
typedef pinocchio::ModelTpl<Scalar> pinModel;
```

Reply:
pinocchio elements are used once or twice in a file. Defining them in the class scope, just to use them once, is not necessary, and does nothing to simplify the readability of the code.